src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php

```diff
@@ -545,6 +545,11 @@ private function registerTemplatingConfiguration(array $config, $ide, ContainerB
 
             $container->setParameter('templating.helper.form.resources', $config['form']['resources']);
 
+            $container->getDefinition('fragment.renderer.hinclude')
+                ->addTag('kernel.fragment_renderer', array('alias' => 'hinclude'))
+                ->replaceArgument(0, new Reference('templating'))
```

Comment:
I see an issue here: if someone does not register the PHP engine but registers its own engine (Smarty for instance), the hinclude renderer will not be defined. IMO, this setup should always be set in the XML file, and the Twig compiler pass should only deal with optimizing the argument when possible